python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py

```diff
@@ -261,140 +245,143 @@ def __init__(self, G, backend: str = "torch", renumber_graph: bool = None):
-            # there is a dask partitioning issue.
-            self.__graph.renumber_edges_by_type(prev_id_column=self.__old_edge_col_name)
+    def __construct_graph(self, edge_info, multi_gpu=False) -> cugraph.MultiGraph:
+        # Ensure the original dict is not modified.
+        edge_info_cg = {}
+        for pyg_can_edge_type in sorted(edge_info.keys()):
+            src_type, _, dst_type = pyg_can_edge_type
+            srcs, dsts = edge_info[pyg_can_edge_type]
 
-        else:
-            self.__vertex_type_offsets = {}
-            self.__vertex_type_offsets["stop"] = [
-                self.__graph.get_num_vertices(vt) for vt in self.__graph.vertex_types
+            src_offset = np.searchsorted(self.__vertex_type_offsets["type"], src_type)
+            srcs_t = srcs + int(self.__vertex_type_offsets["start"][src_offset])
+
+            dst_offset = np.searchsorted(self.__vertex_type_offsets["type"], dst_type)
+            dsts_t = dsts + int(self.__vertex_type_offsets["start"][dst_offset])
+
+            edge_info_cg[pyg_can_edge_type] = (srcs_t, dsts_t)
+
+        na_src = np.concatenate(
+            [
+                edge_info_cg[pyg_can_edge_type][0]
+                for pyg_can_edge_type in sorted(edge_info_cg.keys())
             ]
-            if self.__backend == "cupy":
-                self.__vertex_type_offsets["stop"] = cupy.array(
-                    self.__vertex_type_offsets["stop"]
-                )
-            else:
-                self.__vertex_type_offsets["stop"] = torch.tensor(
-                    self.__vertex_type_offsets["stop"]
+        )
+
+        na_dst = np.concatenate(
+            [
+                edge_info_cg[pyg_can_edge_type][1]
+                for pyg_can_edge_type in sorted(edge_info_cg.keys())
+            ]
+        )
+
+        na_etp = np.concatenate(
+            [
+                np.array(
+                    [i]
+                    * int(
+                        self.__edge_type_offsets["stop"][i]
+                        - self.__edge_type_offsets["start"][i]
+                        + 1
+                    ),
+                    dtype="int32",
                 )
-                if torch.has_cuda:
-                    self.__vertex_type_offsets["stop"] = self.__vertex_type_offsets[
-                        "stop"
-                    ].cuda()
-
-            cumsum = self.__vertex_type_offsets["stop"].cumsum(0)
-            self.__vertex_type_offsets["start"] = (
-                cumsum - self.__vertex_type_offsets["stop"]
+                for i in range(len(self.__edge_type_offsets["start"]))
+            ]
+        )
+
+        df = pandas.DataFrame(
+            {
+                "src": na_src,
+                "dst": na_dst,
+                # FIXME use the edge type property
+                # "w": np.zeros(len(na_src)),
+                "w": na_etp,
+                "eid": np.arange(len(na_src)),
+                "etp": na_etp,
+            }
+        )
```

Comment:
Does this mean that you always have to create a pandas dataframe for using PyG. I think we should allow a path to provide cudf and dask_cudf dataframes directly as that will probably be much more scalable.

Reply:
I have a separate issue to allow passing in a cudf or dask_cudf dataframe.  That won't be part of this PR.

Reply:
See #3176